karpenter-operator/controllers/karpenter/karpenter_controller.go

```diff
@@ -208,7 +214,70 @@ func (r *Reconciler) reconcileCRDs(ctx context.Context, onlyCreate bool) error {
 	return nil
 }
 
-func (r *Reconciler) reconcileEC2NodeClassOwnedFields(ctx context.Context, userDataSecret *corev1.Secret) error {
+func (r *Reconciler) reconcileVAP(ctx context.Context) error {
```

Comment:
After https://github.com/openshift/hypershift/pull/5404 makes it in. Lets's include some validation for this vap ux.